assets/js/password-requirements.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * @file
+ * Password constraint validation.
+ */
+
+(function ($) {
+
+  // Plugin storage.
+  var plugins = {};
+
+  // Refreshing the fields with ajax causes call attach callback multiple times,
+  // to prevent it, keep track whether it was attached already or not.
+  var attached = false;
+
+  Drupal.behaviors.passwordEnhancementsRequirements = {
+    attach: function (context, settings) {
+      if (!attached) {
+        attached = true;
+
+        // Always reset the ajaxComplete trigger, otherwise it would be bounded
+        // every time when the behaviour being attached.
+        $(window).unbind('ajaxComplete', ajaxComplete).ajaxComplete(ajaxComplete);
+
+        // Add event listener for registering the password constraint plugins.
+        window.addEventListener('passwordEnhancementsPluginLoad', passwordEnhancementsPluginLoad);
+
+        // Add password field event listeners.
+        $('input[name="pass[pass1]"]').keyup(passwordFieldKeyUp);
```

Comment:
The keyup runs after you release the key, change runs when the textfield loses focus, I wanted to do live update on the requirements.
When providing a new password I don't expect an autofill event, so it should be okay I guess, pasting (ctrl+v, cmd+v) the password in the field also triggers keyup.

FYI I haven't used keydown or keypress events either because if the validation happens during while you are pressing the button then it may not update the requirements properly.